src/ParallelTestRunner.jl

```diff
@@ -494,137 +635,71 @@ Several keyword arguments are also supported:
 ## Examples
 
 ```julia
-# Run all tests with default settings
+# Run all tests with default settings (auto-discovers .jl files)
 runtests(MyModule, ARGS)
 
 # Run only tests matching "integration"
 runtests(MyModule, ["integration"])
 
-# Run with custom filter function
-runtests(MyModule, ARGS; test_filter = test -> occursin("unit", test))
-
-# Use custom test record type
-runtests(MyModule, ARGS; RecordType = MyCustomTestRecord)
+# Define a custom test suite
+testsuite = Dict(
+    "custom" => quote
+        @test 1 + 1 == 2
+    end
+)
+runtests(MyModule, ARGS; testsuite)
+
+# Customize the test suite
+testsuite = find_tests(pwd())
+args = parse_args(ARGS)
+if filter_tests!(testsuite, args)
+    # Remove a specific test
+    delete!(testsuite, "slow_test")
+end
+runtests(MyModule, args; testsuite)
 ```
 
 ## Memory Management
 
 Workers are automatically recycled when they exceed memory limits to prevent out-of-memory
 issues during long test runs. The memory limit is set based on system architecture.
 """
-function runtests(mod::Module, ARGS; test_filter = Returns(true), RecordType = TestRecord,
-                  custom_tests::Dict{String, Expr}=Dict{String, Expr}(), init_code = :(),
-                  test_worker = Returns(nothing), stdout = Base.stdout, stderr = Base.stderr)
+function runtests(mod::Module, args::ParsedArgs;
+                  testsuite::Dict{String,Expr} = find_tests(pwd()),
```

Comment:
I think `pwd()` is problematic: https://github.com/JuliaAstro/BackgroundMeshes.jl/pull/19#issuecomment-3447862183.  I don't see a better alternative though, basically we'd like a call-place `@__DIR__`.  Since we're passing the module, one option could be `joinpath(pkgdir(mod), "test")`, but that assumes that (1) tests are _always_ in the `test` subdirectory and that (2) `pkgdir(mod)` return something sensible, but it can also return `nothing` in some cases.